spec/services/true_layer/api_client_mock_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'rails_helper'
+
+RSpec.describe TrueLayer::ApiClientMock do
+  subject { described_class.new(SecureRandom.hex) }
+
+  describe '#provider' do
+    it 'returns sample data' do
+      expect(subject.provider.value).to eq(TrueLayer::SampleData::PROVIDERS)
+    end
+  end
+
+  describe '#account_holders' do
+    it 'returns sample data' do
+      expect(subject.account_holders.value).to eq(TrueLayer::SampleData::ACCOUNT_HOLDERS)
+    end
+  end
+
+  describe '#accounts' do
+    it 'returns sample data' do
+      expect(subject.accounts.value).to eq(TrueLayer::SampleData::ACCOUNTS)
+    end
+  end
+
+  describe '#transactions' do
+    let(:result) { subject.transactions.value }
+
+    it 'returns sample data' do
+      expect(result).not_to be_empty
+      expect(result.pluck(:transaction_id).compact).not_to be_empty
+      expect(result.pluck(:description).compact).not_to be_empty
+      expect(result.pluck(:currency).compact).not_to be_empty
+      expect(result.pluck(:amount).compact).not_to be_empty
+      expect(result.pluck(:timestamp).compact).not_to be_empty
+      expect(result.pluck(:transaction_type).compact).not_to be_empty
+    end
+
+    it 'always returns the same data' do
+      second_result = subject.transactions.value
+      expect(result).to eq(second_result)
+    end
+
+    it 'returns unique transaction_ids' do
+      transaction_ids = result.pluck(:transaction_id)
+      expect(transaction_ids.uniq).to eq(transaction_ids)
+    end
+
+    context 'with known data' do
+      let(:csv_file) { 'spec/fixtures/db/sample_data/bank_transactions.csv' }
```

Comment:
I think you can make your sample smaller (if it can do 2 it can do 22), and make them more varied. I'd suggest something like:
```csv
date,description,amount_out,amount_in
01/12/2020,something with some spaces,1234.56,
12/01/2010,simple,,"£1,234.56"
```
So just two but quite different:

1. date that could be parsed incorrectly with a large month, description with spaces, debit in a simple format without limiters
2. date that could be parsed incorrectly with a large day, one word description, credit in a format with `£` and `,`.